packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/service_account/provider.py

```
@@ -0,0 +1,147 @@
+                    result = await self._service_token_function(**self.config.service_token.kwargs)
+
+                    # Handle function return type: str or tuple[str, str]
+                    if isinstance(result, tuple):
+                        service_header, service_token_value = result
+                    else:
+                        service_token_value = result
+
+                    logger.debug("Retrieved service token via dynamic function")
+
+                except Exception as e:
+                    raise RuntimeError(f"Failed to get service token from function: {e}") from e
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)